allocation_manager/service/query.go

```diff
@@ -0,0 +1,228 @@
+package service
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/enfabrica/enkit/lib/logger"
+)
+
+const (
+	OperatorAND string = "AND"
```

Comment:
I think it's best to add parens.  Otherwise some queries would be awkward to express.  For example,
```
(gen = a0 OR gen = b0) AND gpu = true
```
would have to be written as
```
gen = a0 AND gpu = true OR gen = b0 AND gpu = true
```

Also, AND and OR having different precedence would be surprising to most people.